bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -250,14 +250,102 @@ def _convert_to_vector(args):
 
     return (v, *args[1:])
 
-def _handle_cpp_callables(func, original_template, *args):
+def remove_fn_name_from_signature(signature):
+    """
+    Removes the function name from a signature string.
+    The signature is expected to be in the form of "return_type function_name(type param1, type param2, ...)".
+    """
+    if '(' not in signature or ')' not in signature:
+        raise ValueError(f"Invalid signature format: {signature}")
+
+    return signature[:signature.index(' ')] + signature[signature.index('('):]
+
+def get_cpp_overload_from_templ_proxy(func, types=None):
+    """
+    Gets the C++ overload from a cppyy TemplateProxy using the input
+    and template arguments types.
+    """
+    signature = ", ".join(types) if types else ""
+    template_args = func.__template_args__[1:-1] if func.__template_args__ else ""
+    return func.__overload__(signature, template_args)
+
+def get_column_types(rdf, cols):
+    if rdf is None:
+        return []
+    types = [rdf.GetColumnType(col) for col in cols]
+    return types
+
+def get_overload_based_on_args(overload_types, types):
+    """
+    Gets the C++ overloads for a function based on the provided signatures and types.
+    """
+    if not isinstance(overload_types, dict):
+        raise TypeError("Overload types must be a dictionary.")
+
+    if not isinstance(types, list) or not all(isinstance(t, str) for t in types):
+        raise TypeError("Types must be a list of strings.")
+
+    if len(overload_types) == 1:
+        # If there is only one signature, return it directly
+        return next(iter(overload_types))
+
+    for full_sig, info in overload_types.items():
+        input_types = info.get('input_types', ())
+
+        if len(input_types) != len(types):
+            continue
+
+        match = True
+        for expected, actual in zip(input_types, types):
```

Comment:
So here `actual` comes from `df.GetColumnType`? Why is a loose match enough in this case?